core/src/main/kotlin/org/evomaster/core/output/TestCaseWriter.kt

```diff
@@ -668,12 +694,26 @@ class TestCaseWriter {
         }
     }
 
+    /**
+     * applyEscapes currently sets up the string for printing.
+     * This includes escaping special chars for java and kotlin.
+     * Currently, Strings containing "@" are split, on the assumption (somewhat premature, admittedly) that
+     * the symbol signifies an object reference (which would likely cause the assertion to fail).
+     */
     private fun applyEscapes(string: String): String {
-        val ret = string.replace("\"", "\\\"")
+
+        val timeRegEx = "[0-2]?[0-9]:[0-5][0-9]".toRegex()
+
+        val ret = string.split("@")[0] //first split off any reference that might differ between runs
+                .split(timeRegEx)[0] //split off anything after specific timestamps that might differ
+                .replace("""\\""", """\\\\""")
+                .replace("\"", "\\\"")
                 .replace("\n", "\\n")
                 .replace("\r", "\\r")
 
```

Comment:
would really need some more test cases to check if generated tests can be correctly compiled

Reply:
Agreed, I just haven't gotten around to that yet. Depending on how we will handle the escapes, it may require some more refining as well.